### DISCUSSION TITLE
Is moby affected by etcd vulnerability CVE-2021-28235  ?

### DISCUSSION THREAD

**Discussion by HarinadhD** [moby/moby#47388] [Q&A] [answered]
Moby using etcd 3.5.6 indirectly , etcd 3.5.6 is vulnerable to CVE-2021-28235, Can someone confirm moby affected or not ?

**Comment by HarinadhD** [comment]
CVE-2021-28235 resolved in etcd version 3.5.8

**Reply by thaJeztah** [reply, marked as answer]
Fix for that vulnerability was in https://github.com/etcd-io/etcd/pull/15648, which affects the `v3_server.go` file in the `go.etcd.io/etcd/server/v3/etcdserver` package. That package is not used in our code, and the file / package affected is not part of the code used;

```
tree vendor/go.etcd.io/etcd/server/v3/etcdserver/
vendor/go.etcd.io/etcd/server/v3/etcdserver/
`-- api
    `-- snap
        |-- db.go
        |-- doc.go
        |-- message.go
        |-- metrics.go
        |-- snappb
        |   |-- snap.pb.go
        |   `-- snap.proto
        `-- snapshotter.go

4 directories, 7 files
```

So if your scanner shows this vulnerability, it's a false positive. `govulncheck` also does not show the vulnerability;

```bash
go install golang.org/x/vuln/cmd/govulncheck@latest

govulncheck --version
Go: go1.21.6
Scanner: govulncheck@v1.0.4
DB: https://vuln.go.dev
DB updated: 2024-02-13 18:23:23 +0000 UTC

No vulnerabilities found.

git describe --tags
v25.0.3

govulncheck ./...
Scanning your code and 1181 packages across 195 dependent modules for known vulnerabilities...

No vulnerabilities found.
```
